### ISSUE TITLE
friendly_id breaks ActiveRecord::Base.belongs_to_required_by_default in Rails 5.0.0

### ISSUE THREAD

**Issue by nflorentin** [norman/friendly_id#730]
In Rails 5.0.0, a new config was introduced to activate deactivate the automatic presence validation on **belongs_to** associations.

In config/active_record_belongs_to_required_by_default.rb:
`Rails.application.config.active_record.belongs_to_required_by_default = true`
### Without friendly_id in gemfile

```
> Rails.application.config.active_record.belongs_to_required_by_default
=> true
> ActiveRecord::Base.belongs_to_required_by_default
=> true
> User.belongs_to_required_by_default
=> true
```
### After adding friendly_id in gemfile

In rails console:

```
> Rails.application.config.active_record.belongs_to_required_by_default
=> true
> ActiveRecord::Base.belongs_to_required_by_default
=> nil
> User.belongs_to_required_by_default
=> nil
```

**Comment by nflorentin** [comment]
As a temporary fix, you can replace `Rails.application.config.active_record.belongs_to_required_by_default = true` with `ActiveRecord::Base.belongs_to_required_by_default = true` in config/initializers/active_record_belongs_to_required_by_default.rb but I'm not sure if that could produce any side effect.

**Comment by kimrgrey** [comment]
Hi, @nflorentin! Sounds weird, because, as I know, `friendly_id` doesn't do anything with this config. It looks like a side effect of something. I'll take a look on this issue.

**Comment by sandfoxme** [comment]
@kimrgrey from Rails issue referenced above:

> This is due to initialization order. Any plugin that loads AR will cause it to pull config to AR::Base so subsequent initializers will be too late in the game to take effect.

You may close this one

**Comment by juampi** [comment]
So why was this issue closed? Is there no way FriendlyId can prevent this from happening?

**Comment by edwardmp** [comment]
I agree with @juampi, that this issue is not resolved and friendly_id should be modified to adapt for this and thus the issue shouldn't be considered closed

**Comment by nflorentin** [comment]
Here an example of a similar issue addressed for devise https://github.com/plataformatec/devise/commit/c2c74b0a39238e7d997486814a1c8f75fdaf276f

**Comment by edwardmp** [comment]
@nflorentin great, thanks for linking. Let me try If I can apply this to friendly_id too and submit a PR :)

**Comment by jeffblake** [comment]
Any luck on this?

**Comment by nflorentin** [comment]
The issue is still there in version 5.4.2.

**Comment by nflorentin** [comment]
The issue is still there in version 5.4.2.
